models/cam-fv/camdart_obs_preprocessor.f90

```diff
@@ -0,0 +1,320 @@
+! DART software - Copyright UCAR. This open source software is provided
+! by UCAR, "as is", without charge, subject to all terms of use at
+! http://www.image.ucar.edu/DAReS/DART/DART_download
+
+!> Simple program that opens an obs_seq file and loops over the observations
+!> and copies them to a new output file. This is intended to be a
+!> template for programs that want to alter existing obs in some simple way.
+!>
+!> NOTE:
+!> Observations in a sequence file are processed in ascending time order.
+!> For efficiency at creation time, sequence files support observations in a
+!> different physical order than temporal order.  This program will reorder
+!> the observations to be in temporal order, so the output observations
+!> may have a different observation index number than in the input file.
+
+program camdart_obs_preprocessor
+
+use        types_mod, only : r4, r8, missing_r8, metadatalength
```

Comment:
Was this compiled with options to look for unused variables and procedures?

Reply:
no.  my thought was that it will probably be extended in the future for other uses and keeping possibly future-useful items there might help.  but if you want to remove unused variables now, i'm ok with that too.